geniza/entities/models.py

```diff
@@ -176,14 +177,21 @@ def merge_with(self, merge_people, user=None):
 
         for person in merge_people:
             # ensure any has_page overrides are respected
-            if person.has_page and not self.has_page:
-                self.has_page = True
+            self.has_page = self.has_page or person.has_page
 
-            # ensure missing role and gender are migrated
+            # migrate/copy role and gender if not already present, check for conflicts otherwies
```

Comment:
typo in your comment (Nick used to catch these in mine all the time)
```suggestion
            # migrate/copy role and gender if not already present, check for conflicts otherwise
```